client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -242,13 +252,17 @@ const Checkout = React.createClass( {
 			} );
 		}
 
+		if ( ! selectedSiteSlug ) {
```

Comment:
Actually let's avoid this. If you test the route it mostly uses the receipt id to populate information, and even if we pass in garbage data it will render a basic thank you page.